stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/listener/Detach.java

```diff
@@ -0,0 +1,78 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.messaging.rabbitmq.nativeimpl.listener;
+
+import com.rabbitmq.client.Channel;
+import org.ballerinalang.jvm.scheduling.Strand;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.messaging.rabbitmq.RabbitMQConstants;
+import org.ballerinalang.messaging.rabbitmq.RabbitMQUtils;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.ArrayList;
+
+/**
+ * Detach services from the listener.
+ *
+ * @since 1.0.0
+ */
+@BallerinaFunction(
+        orgName = RabbitMQConstants.ORG_NAME,
+        packageName = RabbitMQConstants.RABBITMQ,
+        functionName = "detach",
+        receiver = @Receiver(type = TypeKind.OBJECT,
+                structType = RabbitMQConstants.LISTENER_OBJECT,
+                structPackage = RabbitMQConstants.PACKAGE_RABBITMQ)
+)
+public class Detach {
+    private static final PrintStream console;
+
+    public static Object detach(Strand strand, ObjectValue listenerObjectValue, ObjectValue service) {
+        ObjectValue channelObject = (ObjectValue) listenerObjectValue.get(RabbitMQConstants.CHANNEL_REFERENCE);
+        Channel channel = (Channel) channelObject.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+        ArrayList<ObjectValue> startedServices =
+                (ArrayList<ObjectValue>) listenerObjectValue.getNativeData(RabbitMQConstants.STARTED_SERVICES);
+        @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+        ArrayList<ObjectValue> services =
+                (ArrayList<ObjectValue>) listenerObjectValue.getNativeData(RabbitMQConstants.CONSUMER_SERVICES);
+        String serviceName = service.getType().getName();
+        String queueName = (String) service.getNativeData(RabbitMQConstants.ALIAS_QUEUE_NAME);
+        try {
+            channel.basicCancel(serviceName);
+            console.println("[ballerina/rabbitmq] Consumer service unsubscribed from the queue " + queueName);
+        } catch (IOException e) {
+            return RabbitMQUtils.returnErrorValue("Error occurred while detaching the service");
+        }
+        ;
+        listenerObjectValue.addNativeData(RabbitMQConstants.CONSUMER_SERVICES,
+                RabbitMQUtils.removeFromList(services, service));
+        listenerObjectValue.addNativeData(RabbitMQConstants.STARTED_SERVICES,
+                RabbitMQUtils.removeFromList(startedServices, service));
```

Comment:
Do we need to add list objects as native data again?

Reply:
I added it to avoid starting consumer services that have already been started, as the user can call the start function multiple times.
`checkpanic channelListener.__start();`